packages/core/contracts/merkle-distributor/implementation/MerkleDistributor.sol

```diff
@@ -156,7 +156,6 @@ contract MerkleDistributor is Ownable {
                 merkleWindows[claims[nextI].windowIndex].rewardToken != currentRewardToken
                 // Next claim reward token is different than current one.
             ) {
-                merkleWindows[_claim.windowIndex].remainingAmount -= batchedAmount;
```

Comment:
This is the bug. `remainingAmount` should be decremented for every claim processed, and now _verifyAndMarkClaimed` takes care of this